src/index.d.ts

```diff
@@ -342,7 +342,7 @@ declare module "redux-persist/es/persistReducer" {
     import { PersistState, PersistConfig } from "redux-persist/es/types";
     // persistReducer
     export interface PersistPartial { _persist: PersistState }
-    export type BaseReducer<S, A> = (state: S | void, action: A) => S;
+    export type BaseReducer<S, A> = (state: S, action: A) => S;
```

Comment:
I think this should probably be `(state: S | undefined, action: A) => S` in order to match up with the redux 4.0 types. After all, the reducer is initialized by passing `undefined` into the reducer function.